### PR TITLE
feat: foxy rebase history

### PR DESCRIPTION
**Pull request by toshiSat**

### CODE REVIEW

File: packages/investor-foxy/src/api/api.ts [hunk 1]
```diff
@@ -808,4 +809,52 @@ export class FoxyApi {
...
+          throw new Error(`Failed to get balance of address at block ${e}`)
+        }
+
+        let timestamp
+        try {
+          timestamp = (await this.web3.eth.getBlock(event.blockNumber)).timestamp
+        } catch (e) {
+          throw new Error(`Failed to get timestamp of block ${e}`)
+        }
+
```
Comment (0xdef1cafe) [inline packages/investor-foxy/src/api/api.ts:853]: do you want to use the const assignment IIFE pattern here?

File: packages/investor-foxy/src/api/api.ts [hunk 2]
```diff
@@ -808,4 +809,52 @@ export class FoxyApi {
...
+  async getRebaseHistory(input: BalanceInput) {
+    const { tokenContractAddress, userAddress } = input
+    this.verifyAddresses([tokenContractAddress])
+
+    const contract = new this.web3.eth.Contract(foxyAbi, tokenContractAddress)
+    const fromBlock = 14381454 // genesis rebase
+
+    const rebaseEvents = (
+      await contract.getPastEvents('LogRebase', {
```
Comment (0xdef1cafe) [inline packages/investor-foxy/src/api/api.ts:821]: try catch?

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]
see comments

**Review by elmutt — APPROVED** [review]

File: packages/investor-foxy/src/api/api.ts (hunk 1 above)
Comment (toshiSat) [reply packages/investor-foxy/src/api/api.ts:853]: sure

File: packages/investor-foxy/src/api/api.ts [hunk 3]
```diff
@@ -808,4 +809,69 @@ export class FoxyApi {
...
+          } catch (e) {
+            throw new Error(`Failed to get timestamp of block ${e}`)
+          }
+        })()
+
+        return { balance, timestamp }
+      })
+    )
+
+    return rebaseChartData
```
Comment (0xdef1cafe) [inline packages/investor-foxy/src/api/api.ts:875]: do an `.every` on rebaseChartData to check for rejected promises, log an error, and return an empty array

File: packages/investor-foxy/src/api/api.ts [hunk 4]
```diff
@@ -808,4 +809,69 @@ export class FoxyApi {
...
+        blockNumber,
+        returnValues: { epoch }
+      } = rebaseEvent
+      return {
+        blockNumber,
+        epoch
+      }
+    })
+
+    const rebaseChartData = await Promise.all(
```
Comment (0xdef1cafe) [inline packages/investor-foxy/src/api/api.ts:845]: use Promise.allSettled

File: packages/investor-foxy/sample.env [hunk 5]
```diff
@@ -1 +1,2 @@
-MNEMONIC='match with local foxy (only used in cli)'
\ No newline at end of file
+MNEMONIC='match with local foxy (only used in cli)'
+ARCHIVE_NODE=https://link-to-archive-node.com
```
Comment (0xApotheosis) [inline packages/investor-foxy/sample.env:2]: We don't enforce this via lint (though probably should) - terminate this line with a new line.

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**